File: demoRDLCvb/demoRDLCvb/Form1.vb
Imports System.Data.SqlClient
Imports Microsoft.Reporting.WinForms
Public Class Form1

    Private Sub Form1_Load(sender As Object, e As EventArgs) Handles MyBase.Load
    End Sub

    Private Sub btnGenerate_Click(sender As Object, e As EventArgs) Handles btnGenerate.Click
        Dim objConn As SqlConnection
        Dim objCmd As SqlCommand
        Dim objReader As SqlDataReader
        Dim objDataset As DataSet = New cusDS
        Dim strSQL As String
        Dim DB_CONN_STRING As String

        DB_CONN_STRING = "Initial Catalog=QLCB;" & _
                                            "Data Source=.\FBOSQL;Integrated Security=SSPI;"

        strSQL = ""
        strSQL = strSQL & "SELECT * FROM KHACHHANG"

        objConn = New SqlConnection(DB_CONN_STRING)

        objCmd = New SqlCommand(strSQL, objConn)

        objCmd.CommandType = CommandType.Text
        objCmd.Connection.Open()
        objReader = objCmd.ExecuteReader
        objDataset.Tables(0).Clear()
        objDataset.Tables(0).Load(objReader)
        objReader.Close()
        objConn.Close()
        Dim rds As ReportDataSource = New ReportDataSource
        ' Note: "dsSalesRpt_dtSalesRpt" was name we gave to the report dataset
        ' when setting it up with the wizard

        ' Tên myCus này trùng với tên lúc connect đến table => phải nhớ
        rds.Name = "myCus"
        rds.Value = objDataset.Tables(0)
        Dim paramList As New Generic.List(Of ReportParameter)
        paramList.Add(New ReportParameter("title"))
        paramList.Add(New ReportParameter("tenkh"))
        paramList.Add(New ReportParameter("makh"))
        paramList.Add(New ReportParameter("dchi"))
        paramList.Add(New ReportParameter("dthoai"))
        'paramList.Add(New ReportParameter("prmToDate", Format(dtpToDate.Value, "M/d/yyyy")))
        With frmReportViewer
            ' Note: When assigning the ReportEmbeddedResource in the statement below, you must
            ' qualify the report object name with the project name (in this case "ReportExample")
            .rptView.LocalReport.ReportEmbeddedResource = "demoRDLCvb.cuscusRpt.rdlc" 'demoRDLCvb.cuscusRpt.rdlc => (tên project).(tên file report).rdlc
            .rptView.LocalReport.DataSources.Clear()
            .rptView.LocalReport.DataSources.Add(rds)
            .rptView.LocalReport.SetParameters(paramList)
            .rptView.RefreshReport()
            .Text = "Customer Report" ' Tên của form report
            .ShowDialog()

        End With


    End Sub

    Private Sub btnExit_Click(sender As Object, e As EventArgs) Handles btnExit.Click
        Me.Close()
    End Sub
End Class

File: RDLC/RDLC/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using System.Data.SqlClient;
using Microsoft.Reporting.WinForms;

namespace RDLC
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void Form1_Load(object sender, EventArgs e)
        {

            this.rpvBaoCao.RefreshReport();
        }

        private void btnBaoCao_Click_1(object sender, EventArgs e)
        {
            //Khai báo câu lệnh SQL
            String sql = "Select * from khachhang where makh > '0015'";
            SqlConnection con = new SqlConnection();
            //Truyền vào chuỗi kết nối tới cơ sở dữ liệu
            //Gọi Application.StartupPath để lấy đường dẫn tới thư mục chứa file chạy chương trình 
            con.ConnectionString = @"Data Source=NAM\FBOSQL;Initial Catalog=QLCB;Persist Security Info=True;User ID=Nam;Password=Prayer@1";
            SqlDataAdapter adp = new SqlDataAdapter(sql, con);
            DataSet ds = new DataSet();
            adp.Fill(ds);
            //Khai báo chế độ xử lý báo cáo, trong trường hợp này lấy báo cáo ở local
            rpvBaoCao.ProcessingMode = Microsoft.Reporting.WinForms.ProcessingMode.Local;
            //Đường dẫn báo cáo
            rpvBaoCao.LocalReport.ReportPath = "CustomerList.rdlc";
            //Nếu có dữ liệu
            if (ds.Tables[0].Rows.Count > 0)
            {
                //Tạo nguồn dữ liệu cho báo cáo
                ReportDataSource rds = new ReportDataSource();
                rds.Name = "Customer";
                rds.Value = ds.Tables[0];
                //Xóa dữ liệu của báo cáo cũ trong trường hợp người dùng thực hiện câu truy vấn khác
                rpvBaoCao.LocalReport.DataSources.Clear();
                //Add dữ liệu vào báo cáo
                rpvBaoCao.LocalReport.DataSources.Add(rds);
                //Refresh lại báo cáo
                rpvBaoCao.RefreshReport();
            }
        }
    }
}